lz4_bin/src/main.rs

```diff
@@ -2,103 +2,154 @@ use anyhow::Result;
 use argh::FromArgs;
 
 use std::fs::File;
-use std::io;
-use std::path::Path;
-use std::path::PathBuf;
+use std::io::{self, Write};
+use std::path::{Path, PathBuf};
 
 #[derive(FromArgs, Debug)]
-/// Reach new heights.
+/// [De]Compress data in the lz4 format.
 struct Options {
-    //#[argh(option, default = "false")]
     #[argh(switch)]
     /// delete original files (default: false)
     clean: bool,
 
+    #[argh(switch, short = 'f')]
+    /// overwrite output files
+    force: bool,
+
     #[argh(switch, short = 'd')]
     /// force decompress
     decompress: bool,
 
-    // #[argh(switch, short = 'f')]
-    // /// overwrite_files
-    // force: bool,
     #[argh(positional)]
+    /// file to compress/decompress
     input_file: Option<PathBuf>,
     //#[argh(positional)]
-    /// zoo
+    /// output file to write to. defaults to stdout
     #[argh(option, short = 'o')]
     out: Option<PathBuf>,
-    /// file[s] to compress/decompress
-    /// list of input files
-    #[argh(option, short = 'f')]
-    files: Vec<PathBuf>,
 }
-const LZ_ENDING: &'static str = "lz4";
+const LZ_ENDING: &str = "lz4";
+const LZ_EXTENSION: &str = ".lz4";
 
 fn main() -> Result<()> {
     let opts: Options = argh::from_env();
 
-    if opts.input_file.is_none() && opts.files.is_empty() {
+    let input_file = opts.input_file.filter(|f| f.as_os_str() != "-");
```

Comment:
what does the "-" filter do?

Reply:
If it's `-` if uses stdio, so if it's `None` either because it wasn't passed or `-` was passed then it uses stdin or else it used the filename

Reply:
Why is `None` not enough? I think this would need to be documented

Reply:
It's sort of standard I guess - I was looking at how the `lz4` command does it, and that allows like `foo | lz4 - - | bar`

Reply:
ah yes, I also saw that, but didn't understand it completely. I think one major difference is that here `out` is not a positional parameter. Doesn't hurt so we can keep it for now